frontend/src/components/tab/tab.tsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable react-refresh/only-export-components */
+import styled from '@emotion/styled'
+import {createContext, PropsWithChildren, useContext, useState} from 'react'
+
+interface TabContext {
+  value: number
+  setValue: React.Dispatch<React.SetStateAction<number>>
+}
+
+const tabContext = createContext<TabContext | null>(null)
+
+interface ProviderProps {
+  defaultValue: number
+}
+
+function TabProvider({defaultValue, children}: PropsWithChildren<ProviderProps>) {
+  const [value, setValue] = useState(defaultValue)
+  return <tabContext.Provider value={{value, setValue}}>{children}</tabContext.Provider>
```

Comment:
```suggestion
  return (
    <TabContext.Provider value={value}>
      <TabSetContext.Provider value={setValue}>{children}</TabSetContext.Provider>
    </TabContext.Provider>
  )
```

value 와 setValue를 나눠서 Provider 를 작성하면 렌더링 최적화가 가능해요!

Reply:
이건 생각지 못했네요!